### ISSUE TITLE
Regarding Distance Threshold Optimization

### ISSUE THREAD

**Issue by hsfzxjy** [wangguanan/light-reid#33]
Hi, I've found that DTO requires query and gallery pids as inputs ([code here](https://github.com/wangguanan/light-reid/blob/master/lightreid/evaluations/rank/rank_c2f.py#L262)). Does it mean that DTO is using ground truth during evaluation? (which could be an unfair comparison against other methods).

**Comment by wangguanan** [comment]
DTO is only used in the validation stage, not the eval stage.

in the training stage, we first split training data into training and validation sets.
then,
1. train AiO with the training set.
2. process DTO with the validation set.
3. train AiO with the whole training set.
4. eval our CtF model.

So, the comparison is fair with other state-of-the-arts.

**Comment by hsfzxjy** [comment]
@wangguanan OK, I see.

If I am going to use CtF in a generalized ReID settings, that is, data distribution of the training and testing sets are quite different, is the current DTO applicable? Or how to modify it to fit such settings?

Thanks for your suggestions!

**Comment by wangguanan** [comment]
yes, the current DTO is applicable. 
no need to change.

**Comment by hsfzxjy** [comment]
Thank you very much.
